### PR TITLE
hotplug/unplug memory tests update

### PR DESCRIPTION
**Pull request by xutian**
According feature ower's suggestion update test steps.

1. add reboot step in the end of test
2. increase sleep time to hotplug memory when vm paused
3. add two memory object in different numa node.

### CODE REVIEW

**Comment by suqinhuang** [comment]
Hi xutian,
what's the id

File: qemu/tests/hotplug_mem.py [hunk 1]
```diff
@@ -31,40 +31,37 @@ def run_background_test(self):
...
+        if len(mem_devs_origin) > len(mem_devs_post):
+            mem_devs = mem_devs_origin - mem_devs_post
+            vm, operation = post_vm, "hotplug"
+        elif len(mem_devs_origin) < len(mem_devs_post):
+            mem_devs = mem_devs_post - mem_devs_origin
+            vm, operation = pre_vm, "unplug"
+        func = getattr(self, "%s_memory" % operation)
+        map(lambda x: func(vm, x), mem_devs)
+        vm.verify_alive()
+        vm.reboot()
```
Comment (Zhengtong) [inline qemu/tests/hotplug_mem.py]: Hi, @xutian , Is the reboot here necessary ?  As there is a "reboot action" in "finally" part in start_test function

Comment (CongLi) [inline qemu/tests/hotplug_mem.py]: Agree with @Zhengtong .
@xutian 
Could you pelease paste the id here ?

**Comment by xutian** [comment]
ID: 1285321

File: qemu/tests/hotplug_mem.py [hunk 2]
```diff
@@ -31,40 +31,35 @@ def run_background_test(self):
...
         Return memory object and pc-dimm devices by given name
         """
-        devices = []
-        for dtype in ["mem", "dimm"]:
-            dev_qid = "%s-%s" % (dtype, name)
-            devs = vm.devices.get_by_qid(dev_qid)
-            if devs:
-                devices.append(devs[0])
-        return devices
+        dev_ids = map(lambda x: "-".join(x, name), ["mem", "dimm"])
```
Comment (Zhengtong) [inline qemu/tests/hotplug_mem.py]: 00:52:17 ERROR| Traceback (most recent call last):
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/avocado-vt/avocado_vt/test.py", line 169, in runTest
00:52:17 ERROR|     self._runTest()
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/avocado-vt/avocado_vt/test.py", line 311, in _runTest
00:52:17 ERROR|     run_func(self, params, env)
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/avocado-vt/virttest/error_context.py", line 135, in new_fn
00:52:17 ERROR|     return fn(*args, **kwargs)
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem.py", line 108, in run
00:52:17 ERROR|     hotplug_test.start_test()
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem.py", line 68, in start_test
00:52:17 ERROR|     devs = self.get_mem_by_name(vm, target_mem)
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem.py", line 55, in get_mem_by_name
00:52:17 ERROR|     dev_ids = map(lambda x: "-".join(x, name), ["mem", "dimm"])
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem.py", line 55, in <lambda>
00:52:17 ERROR|     dev_ids = map(lambda x: "-".join(x, name), ["mem", "dimm"])
00:52:17 ERROR| TypeError: join() takes exactly one argument (2 given)

**Comment by xutian** [comment]
@Zhengtong  thanks your testing, it's really a typo and it has been fix, can you help to review again.

File: qemu/tests/hotplug_mem.py [hunk 3]
```diff
@@ -31,40 +31,35 @@ def run_background_test(self):
...
+        deivces = filter(None, map(vm.devices.get_by_qid, dev_ids))
```
Comment (Zhengtong) [inline qemu/tests/hotplug_mem.py]: Hi , @xutian , I think here is "devices"  , maybe another typo error.

**Comment by xutian** [comment]
@Zhengtong  done, thanks

**Comment by Zhengtong** [comment]
ACK
Run passed with my test cases

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

PASSED

**Comment by xutian** [comment]
merged, thanks @suqinhuang  and @Zhengtong

**Pull request merged**